File: ar-app-android-project/app/src/main/java/com/example/robotkarolar/karollogic/instructions/statements/RightTurn.java
package com.example.robotkarolar.karollogic.instructions.statements;

import com.example.robotkarolar.karollogic.instructions.Instruction;
import com.example.robotkarolar.karollogic.instructions.visitors.InstructionVisitor;

public class RightTurn extends Instruction {
    @Override
    public void accept(InstructionVisitor visitor) {
        visitor.accept(this);
    }
}

File: ar-app-android-project/app/src/main/java/com/example/robotkarolar/karollogic/instructions/statements/RightTurn.kt
package com.example.robotkarolar.karollogic.instructions.statements

import com.example.robotkarolar.karollogic.instructions.Instruction
import com.example.robotkarolar.karollogic.instructions.visitors.InstructionVisitor
import kotlinx.parcelize.Parcelize

@Parcelize
class RightTurn : Instruction() {
    override fun accept(visitor: InstructionVisitor?) {
        visitor!!.accept(this)
    }
}